mm2src/common/mm_number.rs

```diff
@@ -162,55 +176,87 @@ impl Default for MmNumber {
-        let rhs: BigRational = rhs.clone().into();
-        (lhs / rhs).into()
+        for num in vals {
+            let decimal: BigDecimal = BigDecimal::from_str(num).unwrap();
+            let expected: MmNumber = from_dec_to_ratio(decimal.clone()).into();
+            let actual: MmNumber = json::from_str(&num).unwrap();
+            assert_eq!(expected, actual);
+        }
     }
-}
 
-impl PartialEq for MmNumber {
-    fn eq(&self, rhs: &MmNumber) -> bool {
-        match self {
-            MmNumber::BigDecimal(lhs) => match rhs {
-                MmNumber::BigDecimal(rhs) => lhs == rhs,
-                MmNumber::BigRational(rhs) => lhs == &from_ratio_to_dec(rhs),
-            },
-            MmNumber::BigRational(lhs) => match rhs {
-                MmNumber::BigDecimal(rhs) => &from_ratio_to_dec(lhs) == rhs,
-                MmNumber::BigRational(rhs) => lhs == rhs,
-            }
+    #[test]
+    fn test_mm_number_deserialize_from_ratio() {
+        let vals: Vec<BigRational> = vec![
+            BigRational::from_integer(0.into()),
+            BigRational::from_integer(81516161.into()),
+            BigRational::new(370.into(), 5123.into()),
+            BigRational::new(1742152.into(), 848841.into()),
+        ];
+
+        for num in vals {
+            let serialized = json::to_string(&num).unwrap();
+            let expected: MmNumber = num.into();
+            let actual: MmNumber = json::from_str(&serialized).unwrap();
+            assert_eq!(expected, actual);
         }
     }
-}
 
-#[test]
-fn test_from_dec_to_ratio() {
-    let number: BigDecimal = "11.00000000000000000000000000000000000000".parse().unwrap();
-    let rational = from_dec_to_ratio(number);
-    assert_eq!(*rational.numer(), 11.into());
-    assert_eq!(*rational.denom(), 1.into());
+    #[test]
+    fn test_mm_number_deserialize() {
+        #[derive(Debug, Serialize, Deserialize, PartialEq)]
+        struct Helper {
+            num: MmNumber,
+            nums: Vec<MmNumber>,
+        }
 
-    let number: BigDecimal = "0.00000001".parse().unwrap();
-    let rational = from_dec_to_ratio(number);
-    assert_eq!(*rational.numer(), 1.into());
-    assert_eq!(*rational.denom(), 100000000.into());
+        let data = Helper {
+            num: BigRational::new(1.into(), 10.into()).into(),
+            nums: vec![
+                BigRational::from_integer(50.into()).into(),
+                BigRational::new(1.into(), 1000.into()).into(),
+                BigRational::from_integer(1000000000000i64.into()).into(),
+                BigRational::new(33.into(), 100.into()).into(),
+                BigRational::new(5.into(), 2.into()).into()
+            ],
+        };
 
-    let number: BigDecimal = 1.into();
-    let rational = from_dec_to_ratio(number);
-    assert_eq!(*rational.numer(), 1.into());
-    assert_eq!(*rational.denom(), 1.into());
+        // A JSON input with plenty of whitespace.
+        let json = r#"
```

Comment:
Using json! macro might be a bit more convenient, e.g.:
https://github.com/KomodoPlatform/atomicDEX-API/blob/34dd28dbc82b2f0dd7fcd2d49d920c11b5be3028/mm2src/mm2_tests.rs#L224-L229

Reply:
Replaced